QuadPaySDK/PaymentWidget/PaymentWidget.swift

```diff
@@ -13,24 +13,61 @@ public final class PaymentWidget: UIView {
     
     let stackView = UIStackView()
     
+    private var request: ApiRequest<WidgetDataResource>?
+    
     @objc public var merchantId: String = "" {
         didSet{
-            MerchantService.shared.fetchMerchants(merchantId: merchantId){ (result) in
-                switch result {
-                case .success(_):
-                    self.paymentWidgetHeaderText.actualPaymentWidgetLabelText = "Split your order in 4 easy payments with Welcome Pay (powered by Zip)."
-                    self.paymentWidgetHeaderText.style()
-                    self.timelapseGraphView.depth = 0
-                    self.timelapseGraphView.drawTimelapseGraph()
-                case .failure(_):
-                    self.paymentWidgetHeaderText.actualPaymentWidgetLabelText = "Split your order in 4 easy payments with Zip."
-                    self.timelapseGraphView.depth = 3
-                    DispatchQueue.main.async {
-                        self.paymentWidgetHeaderText.style()
-                        self.timelapseGraphView.drawTimelapseGraph()
+            if(amount != "0"){
+                self.request = GatewayService.instance.fetchWidgetData(merchantId: merchantId) { [weak self]
+                    (result) in
+                    
+                    guard let self = self else {
+                        return
+                    }
+                    
+                    switch result {
+                    case .success(let result):
+                        print(result as Any)
+                        
+                        guard let widgetData = result else {
+                            DispatchQueue.main.async {
+                                self.layout()
+                            }
+                            return
+                        }
+                        
+                        var maxTier: Double = 0
+                        let amountAsFloat  = Double(self.amount) ?? 0.00
+                        
+                        for(_,element) in widgetData.feeTiers.enumerated() {
+                            let tierAmount = element.feeStartsAt
+                            if(tierAmount <= amountAsFloat ){
+                                if(maxTier < tierAmount){
+                                    maxTier = tierAmount
+                                    self.maxFee = element.totalFeePerOrder
+                                }
+                            }
+                        }
+                        
+                        self.timelapseGraphView.maxFee = self.maxFee
+                        
+                        
+                        self.applyFeeTiers = true
```

Comment:
Is `applyFeeTiers` required?  Can't we just use `maxFee` to determine if we have a fee to apply?